mlflow/gateway/providers/anthropic.py

```diff
@@ -22,6 +22,11 @@ def __init__(self, config: RouteConfig) -> None:
 
     async def completions(self, payload: completions.RequestPayload) -> completions.ResponsePayload:
         payload = jsonable_encoder(payload, exclude_none=True)
+
+        # Set default values if not set at call time
+        payload.setdefault("temperature", 0.0)
+        payload.setdefault("candidate_count", 1)
```

Comment:
shall we move these to base class ?

Reply:
I don't think we can. `MLflowProvider` doesn't need this operation.

Reply:
A bit confused about why we need to do this. Models with `params` signatures will just ignore params that aren't consumed by the model, and MLflow Model Serving provider is only compatible with models that have `params` signatures. So, it seems fine to use the `temperature` and `candidate_count` defaults in the MLflow OSS provider and forward them on to the model. Lmk if I'm missing something @BenWilson2

Reply:
It was for the case where a model isn't saved with any params. Providing params at inference time to a model that has a signature applied without params will raise an MLflowException. 
I've reverted this change in this PR though as we should change that in the params implementation to issue a Warning instead of throwing (similar to how we already warn that unspecified params are ignored if they are not part of the configured signature; it isn't consistent that we warn in one place and raise in another for effectively same behavior of ignoring specified parameters).